src/main/java/com/rabbitmq/client/MetricsCollector.java

```diff
@@ -36,6 +36,8 @@
 
     void basicPublish(Channel channel);
 
+    void basicPublishFailure(Channel channel);
```

Comment:
I'd add the thrown exception as a second parameter. Our implementations wouldn't use it, but external ones could.